arctic/store/_ndarray_store.py

```diff
@@ -467,9 +597,48 @@ def _segment_index(self, new_data, existing_index, start, new_segments):
         segments: list of offsets. Each offset is the row index of the
                   the last row of a particular chunk relative to the start of the _original_ item.
                   array(new_data) - segments = array(offsets in item)
-        
+
         Returns:
         --------
         Library specific index metadata to be stored in the version document.
         """
         pass  # numpy arrays have no index
+
+    def _chunked_update(self, arctic_lib, version, symbol, records, ranges, orig_ranges):
+        collection = arctic_lib.get_top_level_collection()
+
+        chunks = [r.tostring() for r in records]
+        lens = [len(i) for i in chunks]
+        chunks = compress_array(chunks)
+
+        seg_count = 0
+        seg_len = 0
+
+        bulk = collection.initialize_unordered_bulk_op()
+        for chunk, rng, orig_rng, rec_len in zip(chunks, ranges, orig_ranges, lens):
+            start = rng[0]
+            end = rng[1]
+            orig_start = orig_rng[0]
+            if orig_start is None:
+                version['up_to'] += rec_len
+                seg_count += 1
+                seg_len += rec_len
+            segment = {'data': Binary(chunk), 'compressed': True}
+            segment['segment'] = start
+            segment['end'] = end
+            sha = checksum(symbol, segment)
+            segment['sha'] = sha
+            if orig_start is None:
+                # new chunk
+                bulk.find({'symbol': symbol, 'sha': sha, 'segment': segment['segment']}
+                          ).upsert().update_one({'$set': segment, '$addToSet': {'parent': version['_id']}})
+            else:
+                bulk.find({'symbol': symbol, 'segment': orig_start}
+                          ).update_one({'$set': segment, '$addToSet': {'parent': version['_id']}})
```

Comment:
If I update a chunk, how do I make the new version doc point to it and not the old one? You'll have two docs with the same segment (i.e. index). The only way i can see to do it would be to add another field to the segment docs (version) and it would have to match a version field in the version doc, so when you read for a given date range, you only read dates in the range that also match the version number of the parent version doc.